redhat-appstudio/infra-deployments
# spi config to secret

spi config moved to secret. you have to manually create it because it contains secret data so we can't commit it.

It also creates Route with generated host.

Issue: https://issues.redhat.com/browse/SVPI-48

to test:
1. deploy everything with `./hack/bootstrap-cluster.sh`
2. you should see spi deployed but pods failing to start due to missing mount
3. update `components/spi/config.yaml` with your values (you need to create GitHub oauth app first)
4. create actual secret in cluster with `kubectl create secret generic oauth-config --from-file=components/spi/config.yaml -n spi-system`
5. in few moments, you should see pods running
6. follow https://github.com/redhat-appstudio/service-provider-integration-operator#go-through-the-oauth-flow-manually to test oauth flow


❗  Links to my fork branch are there for testing purpose. I'll revert them back to infra-deployment repo before the merge
